noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -220,24 +219,24 @@ impl PrivateContext {
         self.side_effect_counter = self.side_effect_counter + 1;
     }
 
-    pub fn request_nullifier_secret_key(&mut self, account: AztecAddress) -> GrumpkinPrivateKey {
-        let key_pair = if self.nullifier_key.is_none() {
-            let key_pair = get_nullifier_key_pair(account);
-            validate_nullifier_key_against_address(account, key_pair.public_key);
-            let request = NullifierKeyValidationRequest { public_key: key_pair.public_key, secret_key: key_pair.secret_key };
+    pub fn request_app_nullifier_secret_key(&mut self, account: AztecAddress) -> Field {
+        let keys = if self.nullifier_key.is_none() {
+            let keys = get_nullifier_keys(account);
+            let request = NullifierKeyValidationRequest {
+                master_nullifier_public_key: keys.master_nullifier_public_key,
+                app_nullifier_secret_key: keys.app_nullifier_secret_key
+            };
             self.nullifier_key_validation_requests.push(request);
-            self.nullifier_key = Option::some(key_pair);
-            key_pair
+            self.nullifier_key = Option::some(keys);
+            keys
```

Comment:
Maybe I misunderstood, but didn't we want to avoid exposing the master_nullifier_public_key to the app code?

Reply:
@Thunkar no, it's the master nullifier **secret** key that can't enter app circuit. All the public keys are a public information and they will be stored in the public storage of registry contract.
<img width="981" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/13470840/da4507fb-f02a-4509-8b79-7f4e2c4e9b39">

Reply:
I am an idiot 🎶 

Carry on, LGTM